### PR TITLE
[redirect]: Move types logic to their own file

### PR DESCRIPTION
**Pull request by erbesharat**
**What this PR does / why we need it**:
Moves types logic to their own files and uses each type's config struct to pass data.

**Special notes for your reviewer**:
We didn't have any plans for this change but I thought it may be good to clear the main `Redirect` function a bit and move each type's logic to its own file. We can see this as a step to the plugin-based system idea that we talked about before.

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by erbesharat** [comment]
One thing that I don't feel good about it is the way the response writer and request are passed to the type instance. Is it better to pass them on each method call or replace instance's writer or request wherever it's needed?

File: path.go [hunk 1]
```diff
@@ -25,11 +25,65 @@ import (
 	"time"
 )
 
+// Path contains the data that are needed to redirect path requests
+type Path struct {
+	rw  http.ResponseWriter
+	req *http.Request
+	c   Config
+
```
Comment (stp-ip) [inline path.go]: Whitespace?

File: txtdirect.go [hunk 2]
```diff
@@ -124,27 +128,31 @@ func isIP(host string) bool {
...
-
-	host := r.Host
-	path := r.URL.Path
-
-	bl := make(map[string]bool)
-	bl["/favicon.ico"] = true
-
-	if bl[path] {
-		redirect := strings.Join([]string{host, path}, "")
+func blacklist(w http.ResponseWriter, r *http.Request, c Config) error {
```
Comment (stp-ip) [inline txtdirect.go]: Nit: Not sure blacklist does describe the function well. Maybe blacklistRedirect? Not sure.

Comment (erbesharat) [reply txtdirect.go]: Can't think of anything better than `blacklistRedirect`. It's clear enough and I'm fine with it.

File: path.go (hunk 1 above)
Comment (erbesharat) [reply path.go]: Oops, missed this one :smile:

**Comment by erbesharat** [comment]
All the tests pass on my machine but something is wrong with the apache servers. 

Logs:
```
go: git.apache.org/thrift.git@v0.0.0-20180902110319-2566ecd5d999: git fetch -f origin refs/heads/*:refs/heads/* refs/tags/*:refs/tags/* in /go/pkg/mod/cache/vcs/83dba939f95a790e497d565fc4418400145a1a514f955fa052f662d56e920c3e: exit status 128:
	fatal: unable to access 'https://git.apache.org/thrift.git/': Failed to connect to git.apache.org port 443: Operation timed out
go: error loading module requirements
go: finding git.apache.org/thrift.git v0.0.0-20180902110319-2566ecd5d999
go: git.apache.org/thrift.git@v0.0.0-20180902110319-2566ecd5d999: git fetch -f https://git.apache.org/thrift.git refs/heads/*:refs/heads/* refs/tags/*:refs/tags/* in /go/pkg/mod/cache/vcs/83dba939f95a790e497d565fc4418400145a1a514f955fa052f662d56e920c3e: exit status 128:
	fatal: unable to access 'https://git.apache.org/thrift.git/': Failed to connect to git.apache.org port 443: Operation timed out
go: error loading module requirements
make: *** [Makefile:19: test] Error 1
make: *** [docker-test] Error 2
```

**Comment by stp-ip** [comment]
I'll merge and with Go1.13 and proxy.golang.org on by default these flakes of deps should be almost removed.

**Pull request merged**